src/interpreter.c

```diff
@@ -20,8 +20,98 @@ typedef struct {
     jl_value_t **locals; // slots for holding local slots and ssavalues
     jl_svec_t *sparam_vals; // method static parameters, if eval-ing a method body
     int preevaluation; // use special rules for pre-evaluating expressions
+    size_t ip; // Points to the currently-evaluating statement
 } interpreter_state;
 
+// Backtrace support;
+#ifdef _OS_LINUX_
+extern uintptr_t __start_jl_enter_interpreter_frame;
+extern uintptr_t __stop_jl_enter_interpreter_frame;
+
+extern uintptr_t __start_jl_interpreter_frame;
+extern uintptr_t __stop_jl_interpreter_frame;
+
+#define SECT_INTERP JL_SECTION("jl_interpreter_frame")
+#define SECT_ENTER_INTERP JL_SECTION("jl_enter_interpreter_frame")
```

Comment:
... I got confused what the linux version is doing here. I assume this is the started but not finished version?

Reply:
Right, haven't tried on linux yet.